packages/clerk-js/src/core/resources/Base.ts

```diff
@@ -67,8 +68,22 @@ export abstract class BaseResource {
     }
 
     if (status >= 400) {
-      throw new ClerkAPIResponseError(statusText, {
-        data: payload?.errors as ClerkAPIErrorJSON[],
+      const errors = payload?.errors as ClerkAPIErrorJSON[];
+      const safeErrorMessage = errors?.[0]?.long_message;
+
+      if (errors?.[0]?.code === 'origin_invalid' && isProductionFromPublishableKey(BaseResource.clerk.publishableKey)) {
+        const prodDomain = BaseResource.clerk.frontendApi.replace('clerk.', '');
+        throw new ClerkAPIResponseError(
+          `Clerk: Production Keys are only allowed for domain "${prodDomain}". \nAPI Error: ${safeErrorMessage}`,
+          {
+            data: errors,
+            status: status,
+          },
+        );
+      }
+
+      throw new ClerkAPIResponseError(safeErrorMessage || statusText, {
```

Comment:
1. `statusText` was empty most of the time (not sure why, `status` was populated)
2. We do have a custom `toString()` for `ClerkAPIResponseError` but that is not guaranteed to be called.